Project/Gem/Source/Vision/IdealVisionSystem.cpp

```diff
@@ -183,7 +192,7 @@ namespace ROS2::Demo
         IdealVisionSystemConfiguration::Reflect(context);
         if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
         {
-            serialize->Class<IdealVisionSystem, ROS2SensorComponent>()->Version(1)->Field(
+            serialize->Class<IdealVisionSystem, ROS2SensorComponentBase>()->Version(1)->Field(
```

Comment:
```suggestion
            serialize->Class<IdealVisionSystem, SensorBaseType>()->Version(1)->Field(
```

Because ROS2SensorComponentBase is a template, when SensorBaseType is an alias for complete type (specialized template) - could possibly not reflect properly.